src/main/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/service/BookerDetailsService.kt

```diff
@@ -0,0 +1,77 @@
+package uk.gov.justice.digital.hmpps.prison.visitbooker.registry.service
+
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.dto.BookerDto
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.dto.CreateBookerDto
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.exceptions.BookerNotFoundException
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.model.entity.Booker
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.model.entity.Prisoner
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.model.entity.Visitor
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.model.repository.BookerPrisonerRepository
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.model.repository.BookerPrisonerVisitorRepository
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.model.repository.BookerRepository
+
+@Service
+class BookerDetailsService(
+  private val bookerRepository: BookerRepository,
+  private val prisonerRepository: BookerPrisonerRepository,
+  private val visitorRepository: BookerPrisonerVisitorRepository,
+) {
+
+  @Transactional
+  fun getBooker(reference: String): Booker {
```

Comment:
Could annotate with @Throws(BookerNotFoundException::class) ?

And could the Transactional be Transactional(readonly = true) ?

Reply:
It might not be readonly, depends on the consumer. It's a RuntimeException